app/download/views.py

```diff
@@ -1,19 +1,29 @@
+from uuid import UUID
+
 from flask import Blueprint, current_app, jsonify, make_response, request, send_file
+from notifications_utils.base64_uuid import base64_to_bytes
 
 from app import document_store
-
 from app.utils.store import DocumentStoreError
 
 download_blueprint = Blueprint('download', __name__, url_prefix='')
 
 
-@download_blueprint.route('/services/<uuid:service_id>/documents/<uuid:document_id>', methods=['GET'])
+@download_blueprint.route('/d/<base64_uuid:service_id>/<base64_uuid:document_id>', methods=['GET'])
```

Comment:
The actual shared URL will point to the frontend app landing page, so this one is only visible after a series of redirects.

So I think it's worth keeping the readable resources (`/services/` and `/documents/`), but I'm not sure what's the best thing to do with IDs. It makes sense for them to match the S3 keys, but at the same time it'd be nice to have the upload - copy the URL - download loop working within the API without any manual conversion.

Reply:
On the other hand, I think whatever we decide here for the service ID should match the upload URL. So I'd argue this should stay the way it was and the returned upload URL should point to the frontend app landing page with base64 IDs, which it will unpack for the API request

Reply:
my pref is probably:

frontend: `/d/base64/base64/?key=base64`
api: `/services/uuid/documents/uuid?key=base64`

i think

Reply:
Yeah, I like that